azurelinuxagent/ga/signature_validation.py

```diff
@@ -89,8 +147,84 @@ def validate_signature(package_path, signature):
 
     except CommandError as ex:
         msg = "Signature validation failed for package '{0}'. \nReturn code: {1}\nError details:\n{2}".format(package_path, ex.returncode, ex.stderr)
-        raise SignatureValidationError(msg=msg, code=ExtensionErrorCodes.PluginPackageExtractionFailed)
+        raise SignatureValidationError(msg=msg)
+
+    except Exception as ex:
+        msg = "Signature validation failed for package '{0}'. Error: {1}".format(package_path, ustr(ex))
+        raise SignatureValidationError(msg=msg)
 
     finally:
         if os.path.isfile(signature_path):
             os.remove(signature_path)
+
+
+def validate_handler_manifest_signing_info(manifest, ext_handler):
+    """
+    For signed extensions, the handler manifest includes a "signingInfo" section that specifies
+    the type, publisher, and version of the extension. During signature validation (after extracting zip package),
+    we check these attributes against the values in the goal state. If there is a mismatch, raise an error.
+
+    :param manifest: HandlerManifest object
+    :param ext_handler: Extension object
+    :raises ManifestValidationError: if handler manifest validation fails
+    """
+    man_signing_info = manifest.data.get("signingInfo")
+    if man_signing_info is None:
+        msg = "HandlerManifest.json does not contain 'signingInfo'"
+        raise ManifestValidationError(msg=msg)
+
+    # Validate extension name (publisher + type). This comparison should be case-insensitive, because CRP ignores case for extension name.
+    gs_publisher, gs_type = ext_handler.name.rsplit(".", 1)
```

Comment:
```suggestion
    publisher, type = ext_handler.name.rsplit(".", 1)
```
(no reference to the goal state)

Reply:
Updated to "ext_publisher" and "ext_type"